marker_lints/tests/uitest.rs

```diff
@@ -2,10 +2,15 @@ use marker_uitest::ui_test::*;
 use std::env;
 
 fn main() -> color_eyre::Result<()> {
-    let mut config = marker_uitest::simple_ui_test_config!("tests/ui", "../target")?;
+    let mut config: Config = marker_uitest::simple_ui_test_config!("tests/ui", "../target")?;
 
     config.dependencies_crate_manifest_path = Some("./Cargo.toml".into());
 
+    // `marker_api::declare_lint` uses the `CARGO_CRATE_NAME` environment value.
+    // Setting it here once, makes sure that it'll be available, during the
+    // compilation of all ui test files.
+    std::env::set_var("CARGO_CRATE_NAME", "marker_lints_uitests");
```

Comment:
This looks like a painful papercut. I don't understand how it works. Why is this `marker_lints_uitests` and not just `marker_lints`? The name `marker_lints_uitests` isn't mentioned anywhere else in this PR. I also don't see this code setup for `marker_uilints`.

Could this be avoided? Or could the workaround be encapsulated inside of `marker_uitest::simple_ui_test_config`?

Reply:
`ui_test` doesn't use Cargo and instead calls the rustc driver directly. This is for a few reasons, as Cargo would need a workspace, `Cargo.toml` file and would also be a bit slower for only compiling a single file. However, this also means that any environment values usually set by Cargo (starting with `CARGO_*`) are not set when the uitest files are compiled. 

This should really only be noticed, if someone writes lints for the `marker_api::declare_lint!` macro. That's why only the uitest of `marker_lints` required this environment value to be set. So I don't really see it as a paper cut.